### PR TITLE
Simplify file upload

### PR DESCRIPTION
**Pull request by ItsAdel**
Created a new reusable file upload component for easy drag and drop or upload button for external links and thumbnails. Users can still manually enter links. Added file upload progress bar at the bottom for better UI indicator

External Link:
image

Thumbnail:
image

### CODE REVIEW

File: templates/vue/src/components/FileUpload.vue [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+              'Content-Type': 'multipart/form-data'
+            },
+            onUploadProgress: function( progressEvent ) {
+              this.uploadPercentage = parseInt( Math.round( ( progressEvent.loaded / progressEvent.total ) * 100 ))
+            }.bind(this)
+          }
+        )
+        .then(response => {
+          this.$emit('input', response.data.data.url)
+        })
```
Comment (narendrasss) [inline templates/vue/src/components/FileUpload.vue]: Let's keep the code styling consistent. Make sure the linter runs before you commit. If it doesn't, let me know and we can figure out why it's not running automatically.

File: templates/vue/src/components/FileUpload.vue [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+  data() {
+    return{
+      uploadPercentage: 0
+    }
+  },
+
+  methods: {
+    uploadFile(event) {
+      var formData = new FormData()
```
Comment (narendrasss) [inline templates/vue/src/components/FileUpload.vue]: Try to use `const` over `var` whenever possible. `var` is not block scoped so other places may counter-intuitively have access to it.

File: templates/vue/src/components/NodeModal.vue [hunk 3]
```diff
@@ -144,17 +144,16 @@
...
+              v-if="node.mediaType === 'url-embed'"
               label="External Link"
             >
-              <b-form-input
-                id="node-embed-media-duration"
+              <file-upload
                 v-model="node.typeData.mediaURL"
-                placeholder="Enter embed link (starting with http)"
-                required
+                input-id="node-embed-media-duration"
```
Comment (wynnset) [inline templates/vue/src/components/NodeModal.vue]: We can probably remove this and the `inputId` prop altogether from the component.

File: templates/vue/src/components/NodeModal.vue [hunk 4]
```diff
@@ -144,17 +144,16 @@
...
+                placeholder-text="Enter embed link (starting with http)"
```
Comment (wynnset) [inline templates/vue/src/components/NodeModal.vue]: Is it not possible to just call this prop `placeholder`?

File: templates/vue/src/components/NodeModal.vue [hunk 5]
```diff
@@ -173,11 +172,10 @@
               </b-form-checkbox>
             </b-form-group>
             <b-form-group v-if="addThumbnail">
-              <b-form-input
-                id="node-image-url"
+              <file-upload
                 v-model="node.imageURL"
-                placeholder="Enter the URL for the thumbnail"
-                required
+                input-id="node-image-url"
```
Comment (wynnset) [inline templates/vue/src/components/NodeModal.vue]: Again, I don't think this prop is necessary and can be removed.

File: templates/vue/src/components/NodeModal.vue [hunk 6]
```diff
@@ -173,11 +172,10 @@
...
+                placeholder-text="Enter the URL for the thumbnail"
```
Comment (wynnset) [inline templates/vue/src/components/NodeModal.vue]: Ditto as per comment on previous instance of this tag.

File: templates/vue/src/components/FileUpload.vue [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+<script>
+import axios from "axios"
+
+export default {
+  name: "file-upload",
+  props: {
+    inputId: {
+      type: String,
+      required: true,
+    },
```
Comment (wynnset) [inline templates/vue/src/components/FileUpload.vue]: We can probably get rid of this prop.

File: templates/vue/src/components/FileUpload.vue [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    placeholderText: {
```
Comment (wynnset) [inline templates/vue/src/components/FileUpload.vue]: Can we change this to just `placeholder`?

File: templates/vue/src/components/FileUpload.vue [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+  display: inline-block;
+  overflow: hidden;
+  vertical-align: middle;
+  width: 100%;
+  position: absolute;
+  left: 0;
+  padding-left: 30px;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+}
```
Comment (wynnset) [inline templates/vue/src/components/FileUpload.vue]: Is this necessary? I tried removing all of this and it worked just fine. Let's try and keep the code minimal.

File: templates/vue/src/components/FileUpload.vue [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+  padding: 10px;
+}
+
+progress[value] {
+  /* Reset the default appearance */
+  -webkit-appearance: none;
+  appearance: none;
+
+  width: 250px;
+  height: 20px;
```
Comment (wynnset) [inline templates/vue/src/components/FileUpload.vue]: Can you reduce the height to 5px and move the progress bar closer to the file input field please?

**Pull request merged**